statsmodels/base/transform.py

```diff
@@ -0,0 +1,222 @@
+        """
+        Back-transforms the Box-Cox transformed data array, by means of the
+        indicated method. The provided argument lmbda should be the lambda
+        parameter that was used to initially transform the data.
+
+        Parameters
+        ----------
+        x : array_like
+            The transformed series.
+        lmbda : float
+            The lambda parameter that was used to transform the series.
+        method : {'naive'}
+            Indicates the method to be used in the untransformation. Defaults
+            to 'naive', which reverses the transformation.
+
+            NOTE: 'naive' is implemented natively, while other methods may be
+            available in subclasses!
+
+        Returns
+        -------
+        y : array_like
+            The untransformed series.
+        """
+        method = method.lower()
+        x = np.asarray(x)
+
+        if method == 'naive':
+            if np.isclose(lmbda, 0.):
+                y = np.exp(x)
+            else:
+                y = np.power(lmbda * x + 1, 1. / lmbda)
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return y
+
+    def _est_lambda(self, x, bounds=(-1, 2), method='guerrero', **kwargs):
+        """
+        Computes an estimate for the lambda parameter in the Box-Cox
+        transformation using method.
+
+        Parameters
+        ----------
+        x : array_like
+            The untransformed data.
+        bounds : tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter. Default (-1, 2).
+        method : {'guerrero', 'loglik'}
+            The method by which to estimate lambda. Defaults to 'guerrero', but
+            the profile likelihood ('loglik') is also available.
+        **kwargs : dict
+            Options dict for the specified method.
+            * For 'guerrero': R (int), the seasonality/grouping parameter. Scale
+              ({'mad', 'sd'}), the dispersion measure. Options (dict), to be
+              passed to the optimizer.
+            * For 'loglik': Options (dict), to be passed to the optimizer.
+
+        Returns
+        -------
+        lmbda : float
+            The lambda parameter.
+        """
+        method = method.lower()
+
+        if len(bounds) != 2:
+            raise ValueError("Bounds of length {0} not understood."
+                             .format(len(bounds)))
+        elif bounds[0] >= bounds[1]:
+            raise ValueError("Lower bound exceeds upper bound.")
+
+        if method == 'guerrero':
+            lmbda = self._guerrero_cv(x, bounds=bounds, **kwargs)
+        elif method == 'loglik':
+            lmbda = self._loglik_boxcox(x, bounds=bounds)
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return lmbda
+
+    def _guerrero_cv(self, x, bounds, R=4, scale='sd', options={'maxiter': 25}):
```

Comment:
This is what Guerrero uses in both (1993) and (2004). What does `statsmodels` use internally in similar reshaping operations?

Since Guerrero also describes this (AFAIR) as the grouping parameter, so perhaps `group_length` if we want to follow the literature?